worktree.go

```diff
@@ -262,6 +262,16 @@ func (w *Worktree) setHEADToBranch(branch plumbing.ReferenceName, commit plumbin
 	return w.r.Storer.SetReference(head)
 }
 
+// unstages a set of files -- equivalent to "git restore --staged <file>..."
+func (w *Worktree) RestoreStaged(files ...string) error {
```

Comment:
I rather have the method `Restore`, with a `RestoreOptions` with `Staged` and `Files` as the fields.

Reply:
I can definitely make a change to the signature. I had files as a non-optional field because it is non-optional in the actual git command 
```
git restore --staged
fatal: you must specify path(s) to restore
```
In the scenario where no files are provided should it just discard all staged changes?

Reply:
The options should have a validation, in case the options are invalid, must return an error

Reply:
I have converted to 
```
type RestoreOptions struct {
	// Marks to restore the content in the index
	Staged bool
	// Marks to restore the content of the working tree
	Worktree bool
	// List of file paths that will be restored
	Files []string
}

func (w *Worktree) Restore(o *RestoreOptions) error 
```
It will throw the same error string as git restore